docs/imds_retry_based_on_errors.md

```diff
@@ -0,0 +1,73 @@
+|----------------------|-----------------------------------------------|---------------------------------------------|-----------------------------------------|
+| **400**             | Bad Request (Invalid Parameters)               | **Do not retry**, fix request               | **No retry**                             |
+| **401**             | Unauthorized                                  | **Do not retry**, check authentication setup | **No retry**                             |
+| **403**             | Forbidden                                     | **Do not retry**, verify permissions       | **No retry**                             |
+| **404**             | IMDS endpoint is updating / Identity Not Found | Retry with Exponential Backoff (max 3 retries) | **1s → 2s → 4s (max 4s)**            |
+| **408**             | Request Timeout                                | Retry with Exponential Backoff (max 3 retries) | **1s → 2s → 4s (max 4s)**            |
+| **410**             | IMDS is undergoing updates                    | Retry every 10 seconds (max 70s / 7 attempts). Log each retry. | **10s → 10s → … (up to 7 attempts)**            |
+| **429**             | IMDS Throttle limit reached                   | Retry with Exponential Backoff (max 3 retries) | **1s → 2s → 4s (max 4s)**            |
+| **504**             | Gateway Timeout                               | Retry with Exponential Backoff (max 3 retries) | **1s → 2s → 4s (max 4s)**            |
+| **5xx**             | Transient service error                        | Retry with Exponential Backoff (max 3 retries) | **1s → 2s → 4s (max 4s)**            |
+
+---
+
+## 2️⃣ Identity Propagation & Special Handling for "Identity Not Found" Errors
+- **Scenario:** When an identity is newly assigned to a VM, it may take time for the IMDS service to recognize the identity.
+- **Exception Handling:**  
+  - If the **IMDS response contains "Identity Not Found"**, retry the request using **exponential backoff**.
+  - **Error Code:** **404 (Identity Not Found)**
+  - Recommended retry sequence: **1s → 2s → 4s** (max 3 retries)
+  - If still failing, log an error and return the failure.
+
+---
+
+## 3️⃣ Summary of the updated retry strategy 
+Below is a summary table showing the retry patterns for each scenario:
+
+| **Scenario**                                              | **Attempts**    | **Delay Pattern**                   |
+|-----------------------------------------------------------|-----------------|-------------------------------------|
+| **5xx, 404 (Identity Not Found), 408/504 (Timeout), 429** | Up to **3**     | **Exponential Backoff**: 1s → 2s → 4s |
+| **410 (IMDS Updates)**                                    | Up to **7**     | **Every 10 seconds** (max 70s total) |
+
+### Key Points
+- **Exponential Backoff** applies to:
```

Comment:
nit: put 5xx below 429 to keep in numerical order

Reply:
updated